test/libevmasm/Assembler.cpp

```diff
@@ -254,6 +255,27 @@ BOOST_AUTO_TEST_CASE(immutable)
 	);
 }
 
+BOOST_AUTO_TEST_CASE(subobject_encode_decode)
+{
+	Assembly assembly;
+
+	Assembly subAsm;
+	shared_ptr<Assembly> _subAsmPtr = make_shared<Assembly>(subAsm);
+
+	Assembly subSubAsm;
+	shared_ptr<Assembly> _subSubAsmPtr = make_shared<Assembly>(subSubAsm);
```

Comment:
`_subAsmPtr` and `_subSubAsmPtr` are not parameters so why the `_`?

Also, I think you don't have to create separate variables. `make_shared()` allocates a new object containing a copy of your local variable anyway.

```suggestion
	auto subAsmPtr = make_shared<Assembly>();
	auto subSubAsmPtr = make_shared<Assembly>();
```